extension/serializer/src/main/kotlin/AvroSerializer.kt

```diff
@@ -172,22 +169,29 @@ class AvroSerializer private constructor(
       converter.canConvert(AvroSingleObjectEncoded::class.java, expectedRepresentation)
 
   override fun <T : Any> serialize(data: Any, expectedRepresentation: Class<T>): SerializedObject<T> = when (data) {
-    is MetaData -> metaDataSerializer.serialize(data, expectedRepresentation)
+    is MetaData -> SerializedMetaData(
+      // transform to avroMetaData and recurse
+      serialize(AvroMetaDataExt.convertToAvro(data), expectedRepresentation).data,
+      expectedRepresentation
+    )
     is GenericData.Record -> genericDataRecordSerializer.serialize(data, expectedRepresentation)
     is SpecificRecordBase -> specificRecordSerializer.serialize(data, expectedRepresentation)
     else -> throw IllegalArgumentException("cannot serialize $data to $expectedRepresentation")
   }
 
-
   override fun <S : Any, T : Any> deserialize(serializedObject: SerializedObject<S>): T? {
     if (SerializedType.emptyType() == serializedObject.type) {
       return null
     }
+    if (SerializedMetaData.isSerializedMetaData(serializedObject) && serializedObject.contentType.isAvroSingleObjectEncoded()) {
```

Comment:
Using different style of check... `SerializedMetaData.isSerializedMetaData` vs `serializedObject.type == something` I would like to have the internal API which does it in a same way...

Reply:
SerializedMetaData is a special type provided by axon so I'd prefer to use the default check.
EmptyType does not have that check provided, so we must do it manually. I'd rather provide the "isEmptyType" check to the axon framework than switching here.

Reply:
I mean this: `serializedObject.contentType.isAvroSingleObjectEncoded()`is your method... The Axon way of doing a check on the object is a comaprison (like in `SerializedType.emptyType() == serializedObject.type`), so instead of providing check functions (hiding the implementation), I'd rather do the same as Axon Checks are doing.... 

It becomes more uniform to the reader IMHO.